libbeat/monitoring/inputmon/input.go

```diff
@@ -40,8 +47,9 @@ import (
 // The returned cancel function *must* be called when the input stops to
 // unregister the metrics and prevent resource leaks.
 //
-// Deprecated. Use NewMetricsRegistry instead.
-func NewInputRegistry(inputType, inputID string, optionalParent *monitoring.Registry) (reg *monitoring.Registry, cancel func()) {
+// Deprecated. use NewMetricsRegistry instead.
+// There are still filebeat v1 inputs and other Beats that use this function.
+func NewDeprecatedMetricsRegistry(inputType, inputID string, optionalParent *monitoring.Registry) (reg *monitoring.Registry, cancel func()) {
```

Comment:
This is a very hard deprecation; it breaks any code that was using it. Can we not just mark it with [`// Deprecated: …` godoc](https://go.dev/doc/comment#deprecations)?

Reply:
As I said in the PR I used this strategy to ensure no usage of it was left analysed as we compile for different platforms and architectures and also to make it as explicit as possible that we'd like to not use it form now on.

The few places that needed to keep using it are updated, nothing is broken.
I'd like to keep it like that, do you see it as a major problem?